azurerm/internal/services/web/resource_arm_function_app_slot.go

```diff
@@ -0,0 +1,752 @@
+		return fmt.Errorf("Error retrieving the AuthSettings for AzureRM Function App Slot %q (Function App %q / Resource Group %q): %+v", id.Name, id.FunctionAppName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("function_app_name", id.FunctionAppName)
+	d.Set("kind", resp.Kind)
+	osType := ""
+	if v := resp.Kind; v != nil && strings.Contains(*v, "linux") {
+		osType = "linux"
+	}
+	d.Set("os_type", osType)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.SiteProperties; props != nil {
+		d.Set("app_service_plan_id", props.ServerFarmID)
+		d.Set("enabled", props.Enabled)
+		d.Set("default_hostname", props.DefaultHostName)
+		d.Set("https_only", props.HTTPSOnly)
+		d.Set("daily_memory_time_quota", props.DailyMemoryTimeQuota)
+		d.Set("outbound_ip_addresses", props.OutboundIPAddresses)
+		d.Set("possible_outbound_ip_addresses", props.PossibleOutboundIPAddresses)
+		d.Set("client_affinity_enabled", props.ClientAffinityEnabled)
+	}
+
+	appSettings := flattenAppServiceAppSettings(appSettingsResp.Properties)
+
+	connectionString := appSettings["AzureWebJobsStorage"]
+
+	// This teases out the necessary attributes from the storage connection string
+	connectionStringParts := strings.Split(connectionString, ";")
+	for _, part := range connectionStringParts {
+		if strings.HasPrefix(part, "AccountName") {
+			accountNameParts := strings.Split(part, "AccountName=")
+			if len(accountNameParts) > 1 {
+				d.Set("storage_account_name", accountNameParts[1])
+			}
+		}
+		if strings.HasPrefix(part, "AccountKey") {
+			accountKeyParts := strings.Split(part, "AccountKey=")
+			if len(accountKeyParts) > 1 {
+				d.Set("storage_account_access_key", accountKeyParts[1])
+			}
+		}
+	}
+
+	d.Set("version", appSettings["FUNCTIONS_EXTENSION_VERSION"])
+
+	dashboard, ok := appSettings["AzureWebJobsDashboard"]
+	d.Set("enable_builtin_logging", ok && dashboard != "")
+
+	delete(appSettings, "AzureWebJobsDashboard")
+	delete(appSettings, "AzureWebJobsStorage")
+	delete(appSettings, "FUNCTIONS_EXTENSION_VERSION")
+	delete(appSettings, "WEBSITE_CONTENTSHARE")
+	delete(appSettings, "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING")
+
+	if err = d.Set("app_settings", appSettings); err != nil {
+		return err
+	}
+	if err = d.Set("connection_string", flattenFunctionAppConnectionStrings(connectionStringsResp.Properties)); err != nil {
+		return err
+	}
+
+	identity := azure.FlattenAppServiceIdentity(resp.Identity)
+	if err := d.Set("identity", identity); err != nil {
+		return fmt.Errorf("Error setting `identity`: %s", err)
+	}
+
+	configResp, err := client.GetConfigurationSlot(ctx, id.ResourceGroup, id.FunctionAppName, id.Name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on AzureRM Function App Configuration %q: %+v", id.Name, err)
+	}
+
+	siteConfig := flattenFunctionAppSiteConfig(configResp.SiteConfig)
```

Comment:
To reuse this flatten func the schema for `SiteConfig` in both resources would need to match or this will panic on the `d.Set`. (`pre_warmed_instance_count` is missing from this new resource). One assumes that slots for Premium plans would still make use of this setting, so perhaps this should be added?

Reply:
I'll add `pre_warmed_instance_count` as well, but based on recommendations for other resources I decided to duplicate the flatten and expand code. It's quit easy to let bugs slip in if both `function_app` and `function_app_slot` reuse this code.